include/pressio/ops/eigen/ops_level2.hpp

```diff
@@ -64,9 +64,12 @@ template <
   >
 ::pressio::mpl::enable_if_t<
      ::pressio::all_have_traits_and_same_scalar<A_type, x_type, y_type>::value
-  && ::pressio::Traits<A_type>::package_identifier == PackageIdentifier::Eigen
-  && ::pressio::Traits<x_type>::package_identifier == PackageIdentifier::Eigen
-  && ::pressio::Traits<y_type>::package_identifier == PackageIdentifier::Eigen
+  && (::pressio::is_native_container_eigen<A_type>::value
+   || ::pressio::is_expression_acting_on_eigen<A_type>::value)
+  && (::pressio::is_vector_eigen<x_type>::value
+   || ::pressio::is_expression_acting_on_eigen<x_type>::value)
+  && (::pressio::is_vector_eigen<y_type>::value
+   || ::pressio::is_expression_acting_on_eigen<y_type>::value)
```

Comment:
@fnrizzi do you think we could also have `is_container_eigen = is_native_container_eigen || is_expression_acting_on_eigen` convenience shortcut (and similar for Kokkos) to get rid of `and`/`or` mixtures in places like this ?

Reply:
i would prefer to wait on that because we need to think abut what it means to be "container eigen".